caproto/_circuit.py

```diff
@@ -321,6 +321,13 @@ def _process_command(self, role, command):
                                       ClearChannelResponse)):
                 self.channels_sid.pop(chan.sid)
                 self.channels.pop(chan.cid)
+                to_remove = []
+                for k, v in self._ioids.items():
+                    if v is chan:
+                        to_remove.apend(k)
```

Comment:
Typo
```suggestion
                        to_remove.append(k)
```

Or perhaps better:
```suggestion
                    to_remove = [k for k, v in self._ioids.items() if v is chan]
```